src/annoylib.h

```diff
@@ -164,8 +148,14 @@ struct Angular {
       return random->flip();
   }
   static inline void create_split(const vector<node*>& nodes, int f, Randomness<T>* random, node* n) {
+    // Sample two random points from the set of nodes
+    // Calculate the hyperplane equidistant from them
+    int i = random->index(nodes.size());
+    int j = random->index(nodes.size());
```

Comment:
It's not always an even split – you could end up having 1 point on one side and n-1 points on the other side. Or what did you mean?

Yes you need to be careful about not consistently separating the same points that are close to each other. That's why you need to build an index with multiple trees. Any given tree might separate two close points, but it's unlikely they are separated in all trees